### PR TITLE
DOC: Improve documentation for Index.where

### PR DESCRIPTION
**Pull request by laymonage**
- [x] closes https://github.com/pandanistas/pandanistas_sprint_jakarta2020/issues/15
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -3953,18 +3953,37 @@ def memory_usage(self, deep: bool = False) -> int:
 
     def where(self, cond, other=None):
         """
-        Return an Index of same shape as self and whose corresponding
-        entries are from self where cond is True and otherwise are from
-        other.
+        Return a copy of self with entries replaced where cond is False.
+
+        The replacement is taken from other.
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: I don't think this whole description is very clear. May be you can simply mention in the short summary this is a conditional. And explain what exactly does in the extended summary.

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -3953,18 +3953,37 @@ def memory_usage(self, deep: bool = False) -> int:
...
         Returns
         -------
-        Index
+        pandas.Index
+            Copy of self with entries replaced from other
+            where the condition is False.
+
+        See Also
+        --------
+        DataFrame.where : Replace values in a DataFrame where the condition is False.
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: You can say "Same method for DataFrame.". If you add `DataFrame.where` you should also add `Series.where` (better to have `Series` first, and `DataFrame` later).

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -3953,18 +3953,37 @@ def memory_usage(self, deep: bool = False) -> int:
...
+
+        Examples
+        --------
+        >>> idx1 = pd.Index([42, 21, 34, 96, 72])
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Do you mind trying to find an example with somehow real data? Using arbitrary numbers makes this difficult to read.

For example, something like:
```python
idx = pandas.Index(['car', 'bike, 'train', 'tractor'])
idx.where(idx.isin(['car', 'train'], 'other')
```

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (laymonage) [reply pandas/core/indexes/base.py]: Okay, I've copied the description from `Series.where` and `DataFrame.where`.

**Comment by datapythonista** [comment]
Thanks for the work on this @laymonage

**Pull request merged**